File: 2022/tom/day-1/main.js
const {readFileSync} = require('fs');

function main() {
  const data = readFile();
  const step1Result = step1(data);
  console.log(step1Result);
  const step2Result = step2(data);
  console.log(step2Result);
}

function readFile() {
  const contents = readFileSync('input.txt', 'utf-8');
  const data = contents.split('\r\n');
  return data;
}

function step1(data)  {
	let topCalories = 0;
	let currentCalories = 0;
	data.forEach(function(line) {
		if (line === "") {
			if (topCalories < currentCalories) {
				topCalories = currentCalories;
			}
			currentCalories = 0;
		} else {
			let calories = parseInt(line);
			currentCalories += calories	;
		}
	
	})
	
	if (topCalories < currentCalories) {
		topCalories = currentCalories;
	}
	return topCalories;
}

function step2(data)  {
	let topCalories = [0,0,0];
	let currentCalories = 0;
	data.forEach(function(line) {
		if (line === "") {
			topCalories = updateTopCalories(topCalories,currentCalories);
			currentCalories = 0;
		} else {
			let calories = parseInt(line);
			currentCalories += calories	;
		}
	}) 
	
	topCalories = updateTopCalories(topCalories,currentCalories);

	var total = 0;
	topCalories.forEach(function(calories) {
		total += calories;
	})
	return total;
}

function updateTopCalories(topCalories, currentCalories){
  topCalories.sort(function(a, b){return a - b});
	if (currentCalories > topCalories[0]) {
		topCalories.shift();
		topCalories.push(currentCalories);
	}
	return topCalories;
}

main();

File: 2022/tom/day-1/python/main.py

import sys

def main() : 
	data = []
	if 'test' in sys.argv:
		data = readFile('../test.txt')
	else:
		data = readFile('../input.txt')

	step1Result = step1(data)
	print ("Step 1: ", step1Result)
	step2Result = step2(data)
	print ("Step 2: ", step2Result)


def readFile(fileName):
  contents = open(fileName, "r")
  data = contents.read().split('\n')
  return data


def step1(data) :
	topCalories = 0
	currentCalories = 0
	for line in data:
		if line == "":
			if (topCalories < currentCalories):
				topCalories = currentCalories
			
			currentCalories = 0
		else: 
			calories = int(line)
			currentCalories += calories	
	
	if topCalories < currentCalories:
		topCalories = currentCalories

	return topCalories

def step2(data):
	topCalories = [0,0,0]
	currentCalories = 0
	for line in data:
		if line == "":
			topCalories = updateTopCalories(topCalories,currentCalories)
			currentCalories = 0
		else:
			calories = int(line)
			currentCalories += calories	
	
	topCalories = updateTopCalories(topCalories,currentCalories)

	total = 0
	for calories in topCalories:
		total += calories

	return total

def updateTopCalories(topCalories, currentCalories):
	topCalories.sort()
	if currentCalories > topCalories[0]:
		topCalories.pop(0)
		topCalories.append(currentCalories)

	return topCalories


if __name__ == '__main__':
    sys.exit(main())